File: 2026/04-April/20-April/solution.js
// 2078. Two Furthest Houses With Different Colors
var maxDistance = function(colors) {
    let n = colors.length;
    let ans = 0;

    for (let i = 0; i < n; i++) {
        if (colors[i] !== colors[0]) {
            ans = Math.max(ans, i);
        } 
        if (colors[i] !== colors[n - 1]) {       
            ans = Math.max(ans, n - 1 - i);
        }
    }

    return ans;
};

File: 2026/04-April/20-April/solution.c
// 2078. Two Furthest Houses With Different Colors
int maxDistance(int* colors, int colorsSize) {
    int ans = 0;

    for (int i = 0; i < colorsSize; i++) {
        if (colors[i] != colors[0]) {
            if (i > ans) ans = i;
        }
        if (colors[i] != colors[colorsSize - 1]) {
            if (colorsSize - 1 - i > ans) ans = colorsSize - 1 - i;
        }
    }

    return ans;
}
